runtime/src/main/java/io/quarkus/hazelcast/client/runtime/graal/ServiceLoaderUtils.java

```diff
@@ -0,0 +1,49 @@
+package io.quarkus.hazelcast.client.runtime.graal;
+
+import com.hazelcast.logging.ILogger;
+
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+
+import static com.hazelcast.internal.nio.IOUtil.closeResource;
+
+final class ServiceLoaderUtils {
+    private ServiceLoaderUtils() {
+    }
+
+    static Set<Target_ServiceDefinition> parse(URL url, ClassLoader classLoader, ILogger logger) {
```

Comment:
1. Could you add javadoc explaining what this method does? Or maybe make this method name more explanatory?
2. Wouldn't it be worth to add some unit tests for his class and other you added?

Reply:
I will add docs around the Graal substitution and explain the problem there as well.

I'm working on tests at the moment. Unit testing is useless, we need an integration test that loads a custom extension via SPI

Reply:
Actually, this is covered by Hasan's test suite